pybossa/cache/users.py

```diff
@@ -348,9 +348,26 @@ def delete_user_pref_metadata(name):
 
 @memoize(timeout=ONE_DAY)
 def get_user_preferences(user_id):
+    user_pref = get_user_pref(user_id)
```

Comment:
I think instead of doing two queries to get the user you can get it once and then detirmine user_pref and email at the same time.

user = get_user_by_id(user_id)
user_pref = user.user_pref or {} if user else {}
user_email = user.email_addr if user else None

Reply:
+1